inngest/_internal/comm_lib/handler.py

```diff
@@ -105,29 +105,13 @@ class CommHandler:
     def __init__(
         self,
         *,
-        api_base_url: typing.Optional[str] = None,
         client: client_lib.Inngest,
         framework: server_lib.Framework,
         functions: list[function.Function],
     ) -> None:
         self._client = client
-
         self._mode = client._mode
-
-        api_base_url = api_base_url or os.getenv(
-            const.EnvKey.API_BASE_URL.value
-        )
-        if api_base_url is None:
-            if self._mode == server_lib.ServerKind.DEV_SERVER:
-                api_base_url = const.DEV_SERVER_ORIGIN
-            else:
-                api_base_url = const.DEFAULT_API_ORIGIN
-
-        try:
-            self._api_origin = net.parse_url(api_base_url)
-        except Exception as err:
-            raise errors.URLInvalidError() from err
-
+        self._api_origin = client.api_origin
```

Comment:
For posterity, TS currently also allows a `baseUrl` to be passed when calling `serve()`. I had earmarked it for removal as we want as much config as possible to be purely on the client, though it may be needed if a developer is re-using a single client in multiple places. 🤔

No action needed here - just commenting to mark that the option may still be valid in some cases.

TS order of precedence:

```mermaid
graph TD
    B{"serve({ baseUrl })?"}
    B -->|Yes| C["Use baseUrl from serve()"]
    B -->|No| D{"INNGEST_API_BASE_URL?"}
    D -->|Yes| E["Use INNGEST_API_BASE_URL"]
    D -->|No| F{"INNGEST_BASE_URL?"}
    F -->|Yes| G["Use INNGEST_BASE_URL"]
    F -->|No| H{"new Inngest({ baseUrl })?"}
    H -->|Yes| I["Use baseUrl from new Inngest()"]
    H -->|No| J{"new Inngest({ isDev })?"}
    J -->|true| K["Use http://localhost:8288"]
    J -->|false| L["Use https://api.inngest.com"]
    J -->|No| M{"INNGEST_DEV?"}
    M -->|1| O["Use http://localhost:8288"]
    M -->|0| N["Use https://api.inngest.com"]
    M -->|"[A_URL]"| P["Use [A_URL]"]
    M -->|None| Q{"isProduction()?"}
    Q -->|False| S["Try http://localhost:8288, fallback to https://api.inngest.com"]
    Q -->|True| R["Use https://api.inngest.com"] 
    Q -->|Undefined| T{"Inferred prod checks<br/>e.g. NODE_ENV"}
    T -->|Dev mode| U["Try http://localhost:8288, fallback to https://api.inngest.com"]
    T -->|Prod mode| V["Default to https://api.inngest.com"]
```